Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/Surveys.kt

```diff
@@ -34,18 +38,39 @@ class Surveys @Inject constructor(
     }
 
     suspend fun requestDetails(type: Type): Survey {
+        val surveyConfig = appConfigProvider.getAppConfig().survey
+        Timber.v("Requested survey: %s", surveyConfig)
+        /* no check here:
+         * if surveyOnHighRisk is not enabled, this use case shouldn't have been started in the first place
+         */
+//        if (type == Type.HIGH_RISK_ENCOUNTER && !surveyConfig.surveyOnHighRiskEnabled) {
+//            throw SurveyException(SurveyException.Type.HIGH_RISK_NOT_ENABLED)
+//        }
+        otpRepo.otpAuthorizationResult?.apply {
+            if (authorized && redeemedAt.toDateTime().monthOfYear() == Instant.now().toDateTime().monthOfYear()) {
+                throw SurveyException(SurveyException.Type.ALREADY_PARTICIPATED_THIS_MONTH)
+            }
+        }
+        val otp = otpRepo.otp ?: otpRepo.generateOTP()
+        val errorCode = surveyServer.authOTP(
```

Comment:
Before authenticating the OTP, we have to first make the device attestation and check if it fulfills all requirements. 

Like in `DataDonationTestFragmentViewModel`: 

![image](https://user-images.githubusercontent.com/10398034/107923156-5f8d1080-6f71-11eb-8da0-05e384b92b82.png)

Reply:
thanks! Added it